client/rest-high-level/build.gradle

```diff
@@ -64,3 +82,32 @@ forbiddenApisMain {
   signaturesURLs += [PrecommitTasks.getResource('/forbidden/http-signatures.txt')]
   signaturesURLs += [file('src/main/resources/forbidden/rest-high-level-signatures.txt').toURI().toURL()]
 }
+
+javadoc {
+  // Bundle javadoc from all bundled projects into this project's javadoc
+  configurations.bundled.dependencies.all { Dependency dep ->
+    Project p = dependencyToProject(dep)
+    if (p != null) {
+      evaluationDependsOn(p.path)
+      source += p.sourceSets.main.allJava
+    }
+  }
+}
+jar {
+  from({configurations.bundled.collect { it.isDirectory() ? it : zipTree(it) }}) {
+    exclude 'META-INF/*'
```

Comment:
Won't this break any SPI used by the "bundled" jar?

Reply:
It seems like this is trying to manually do the work of the shadow plugin?

Reply:
Yes on both counts. I started with the shadow plugin but it fought me and it feels so "big". I'll see about using it though. It absolutely handles SPI properly and this doesn't.